packages/framework/src/Model/NumberSequence/AbstractNumberSequenceRepository.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\NumberSequence;
+
+use Doctrine\DBAL\LockMode;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Exception;
+use Shopsys\FrameworkBundle\Model\NumberSequence\Exception\NumberSequenceNotFoundException;
+
+abstract class AbstractNumberSequenceRepository
+{
+    protected const int ID = 1;
+
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $em
+     */
+    public function __construct(
+        protected readonly EntityManagerInterface $em,
+    ) {
+    }
+
+    /**
+     * @return \Doctrine\ORM\EntityRepository<\Shopsys\FrameworkBundle\Model\Complaint\ComplaintNumberSequence>
```

Comment:
should be like this, right?
```suggestion
     * @return \Doctrine\ORM\EntityRepository<\Shopsys\FrameworkBundle\Model\NumberSequence\AbstractNumberSequence>
```

Reply:
Yes, that was fixed later. I forgot to rebase it in to the same fixup commit.